File: conseil/src/main/java/com/example/conseil/entities/Recipe.java
package com.example.conseil.entities;

import com.example.conseil.enums.RecipeCategory;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.sql.Date;

import java.util.List;

@Entity
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
public class Recipe {
  @Id
  @GeneratedValue(strategy = GenerationType.IDENTITY)
  private Long id;
  @Column(nullable = false)
      private String name;
  @Column(nullable = false)
      private  String description;
  @Column(nullable = false)
      private String ingredients;
  @Column(nullable = false)
      private Date datePublication;
  @Column(nullable = false)
      private String instructions;
    @Lob
    @Column(length = 1000000)
    private byte[] image;
     @Enumerated(EnumType.STRING)
     private RecipeCategory category;

      @OneToMany(fetch = FetchType.EAGER)
  private List<Comment> commentList;

  @OneToMany(fetch = FetchType.EAGER)
  private List<FavoriteRecipe> favoriteRecipeList;

  @ManyToOne(fetch = FetchType.LAZY)
  @JoinColumn(name = "specialist_id")
  private Specialist specialist;


}

File: FrontEndAngular/src/app/models/recipe.ts
import {FavoriteRecipe} from "./favorite-recipe";

export class Recipe {
  id: number;
  name: string;
  description: string;
  ingredients: string;
  datePublication: string; // Representing `LocalDate` as ISO string
  instructions: string;
  image: string;
  category: string; // Enum in backend, string in frontend
  commentList: Comment[]; // List of Comment objects
  favoriteRecipeList: FavoriteRecipe[]; // List of FavoriteRecipe objects
  specialistId: number; // Store the ID of the Specialist

  constructor(
    id: number,
    name: string,
    description: string,
    ingredients: string,
    datePublication: string,
    instructions: string,
    image: string,
    category: string,
    commentList: Comment[],
    favoriteRecipeList: FavoriteRecipe[],
    specialistId: number
  ) {
    this.id = id;
    this.name = name;
    this.description = description;
    this.ingredients = ingredients;
    this.datePublication = datePublication;
    this.instructions= instructions;
    this.image = image;
    this.category = category;
    this.commentList = commentList;
    this.favoriteRecipeList = favoriteRecipeList;
    this.specialistId = specialistId;
  }
}
